src/resources/extensions/quarto/docusaurus/docusaurus_writer.lua

```diff
@@ -51,38 +51,42 @@ local function tabset(node, filter)
   return tabs
 end
 
-function Writer(doc, opts)  
-  local filter
-  filter = {
-    CodeBlock = codeBlock,
-
-    DecoratedCodeBlock = function(node)
-      local el = node.code_block
-      return codeBlock(el, node.filename)
-    end,
-
-    Tabset = function(node)
-      return tabset(node, filter)
-    end,
-
-    Callout = function(node)
-      local admonition = pandoc.List()
-      admonition:insert(pandoc.RawBlock("markdown", "\n:::" .. node.type))
-      if node.title then
-        admonition:insert(pandoc.Header(2, node.title))
-      end
-      local content = node.content
-      if type(content) == "table" then
-        admonition:extend(content)
-      else
-        admonition:insert(content)
-      end
-      admonition:insert(pandoc.RawBlock("markdown", ":::\n"))
-      return admonition
-    end
-  }
-  
-  doc = quarto._quarto.ast.walk(doc, filter)
+quarto._quarto.ast.add_renderer("Tabset", function()
+  return quarto._quarto.format.isDocusaurusOutput()
+end, function(node)
+  return tabset(node)
+end)
+
+quarto._quarto.ast.add_renderer("Callout", function()
+  return quarto._quarto.format.isDocusaurusOutput()
+end, function(node)
+  local admonition = pandoc.Blocks({})
+  admonition:insert(pandoc.RawBlock("markdown", "\n:::" .. node.type))
+  if node.title then
+    admonition:insert(pandoc.Header(2, node.title))
+  end
+  local content = node.content
+  if type(content) == "table" then
+    admonition:extend(content)
+  else
+    admonition:insert(content)
+  end
+  admonition:insert(pandoc.RawBlock("markdown", ":::\n"))
+  return admonition
+end)
+
+quarto._quarto.ast.add_renderer("DecoratedCodeBlock", function()
+  return quarto._quarto.format.isDocusaurusOutput()
+end, function(node)
+  local el = node.code_block
+  return code_block(el, node.filename)
+end)
+
+function Writer(doc, opts)
+  doc = quarto._quarto.ast.walk(doc, {
+    CodeBlock = code_block,
+  })
+  assert(doc ~= nil)
```

Comment:
@cscheid Curious to learn more: Is `assert()` in our Lua code a way to had some checks to error early on some feature ? 
Will that create a good error message ? 

I am not using that currently, but always keen to learn new practice ! (which reading others code is always good for !)

 thanks

Reply:
It's a lua-specific thing I only recentlly learned about. In this case that should really never happen, and if it ever did, the error message won't be particularly helpful. It's there because the lua analyzer in the vs code extension knows about it, so it assumes `doc ~= nil` in the lines following the assertion.

I used to write 

```
if doc ~= nil then
  -- luacov: disable
  internal_error() 
  return
  -- luacov: enable
end
```

(the `luacov` statements are telling the lua code coverage tool that those lines shouldn't be counted)

As you can see, that's a lot more lines for the same effect.

Reply:
> It's there because the lua analyzer in the vs code extension knows about it, so it assumes doc ~= nil in the lines following the assertion.

TIL ! Thanks. I'll try to use more of `assert()` now.